app/models/manageiq/providers/azure/cloud_manager/event_catcher/stream.rb

```diff
@@ -25,7 +39,7 @@ def stop
   #
   def each_batch
     while @collecting_events
-      yield get_events.collect { |e| JSON.parse(e) }
+      yield get_events.collect { |e| e.to_hash }
```

Comment:
Why not go further? `get_events.collect(&:to_hash)`

Reply:
Done!